apps/condo/cypress/objects/helpers.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Similar to cy.visit(url), but will track page loading time.
+ * @param {string} url
+ * @example trackedVisit('/auth/signin/')
+ */
+const trackedVisit: (url) => Cypress.Chainable<Performance> = (url) => {
+    // transforms /auth/signin/ to auth.signin
+    // if empty: then 'index' is chosen as metric name
+    const normalizedUrl = url.replace(/^\/?|\/?$/g, '').replaceAll('/', '.') || 'index'
+
+    return cy.visit(url, {
+        onBeforeLoad: (win) => {
+            win.performance.mark(`${normalizedUrl}.startLoading`)
+        },
+        onLoad: (win) => {
+            win.performance.mark(`${normalizedUrl}.stopLoading`)
+        },
+    }).its('performance').then((p) => {
+        p.measure(`${normalizedUrl}.pageLoad`, `${normalizedUrl}.startLoading`, `${normalizedUrl}.stopLoading`)
+        const measure = p.getEntriesByName(`${normalizedUrl}.pageLoad`)[0]
+
+        cy.log(`[helpers.ts] ${normalizedUrl} loaded in: ${measure.duration}`)
+        cy.task('metrics:histogram', [`${normalizedUrl}.pageload`, measure.duration])
+    })
+}
+
+type Span = {
```

Comment:
What did you mean? Maybe it's better to call it `task` or `scenario`?

Reply:
span is an established term for defining a finished part of the large operation

https://github.com/opentracing/specification/blob/master/specification.md

we have only one problem tho.. our traces are not actual traces :-) but that's another story